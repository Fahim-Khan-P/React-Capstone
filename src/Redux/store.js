import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries';
import countryReducer from './country';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    country: countryReducer,
  },
});

export default store;
