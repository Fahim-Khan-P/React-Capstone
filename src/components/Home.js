/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Navbar from './Navbar';
import Europe from './Europ';
import CountryList from './CountryList';
import { fetchCountries } from '../Redux/countries';

const Home = () => {
  const { countries } = useSelector((state) => state.countries);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCountries());
  }, []);
  return (
    <div className="homeScreen">
      <Navbar />
      <Europe />
      <CountryList data={countries} />
    </div>
  );
};

export default Home;
