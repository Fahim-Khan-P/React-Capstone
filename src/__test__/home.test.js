import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import Home from '../components/Home';
import store from '../Redux/store';

it('It should render the homepage components correctly', () => {
  const homepage = render(
    <Router>
      <Provider store={store}>
        <Home />
      </Provider>
    </Router>,
  );
  expect(homepage).toMatchSnapshot();
});
