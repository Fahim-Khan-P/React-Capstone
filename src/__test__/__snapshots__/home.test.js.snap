// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the homepage components correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homeScreen"
      >
        <div
          class="navBar"
        >
          <div
            class="navbarLeftDiv"
          >
            <a
              href="/"
            >
              <i
                class="fa-solid fa-chevron-left"
              />
            </a>
          </div>
          <h3
            class="navbar-middle"
          >
            Europian Countries
          </h3>
          <div
            class="navbarRightDiv"
          >
            <i
              class="fa-solid fa-microphone"
            />
            <i
              class="fa-solid fa-gear"
            />
          </div>
        </div>
        <div
          class="europeMainDiv"
        >
          <div
            class="imageDiv"
          >
            <img
              alt="Europe"
              class="europeImage"
              src="Europ.png"
            />
          </div>
          <div
            class="europeDatialsDiv"
          >
            <h1
              class="euroh2eTitle"
            >
              Europe
            </h1>
            <p
              class="europePopulation"
            >
              746.4 Million
            </p>
          </div>
        </div>
        <div
          class="CountryListMainDiv"
        >
          <h2
            class="CountryListTitle"
          >
            List of Urope Countries
          </h2>
          <div
            class="CountryList"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homeScreen"
    >
      <div
        class="navBar"
      >
        <div
          class="navbarLeftDiv"
        >
          <a
            href="/"
          >
            <i
              class="fa-solid fa-chevron-left"
            />
          </a>
        </div>
        <h3
          class="navbar-middle"
        >
          Europian Countries
        </h3>
        <div
          class="navbarRightDiv"
        >
          <i
            class="fa-solid fa-microphone"
          />
          <i
            class="fa-solid fa-gear"
          />
        </div>
      </div>
      <div
        class="europeMainDiv"
      >
        <div
          class="imageDiv"
        >
          <img
            alt="Europe"
            class="europeImage"
            src="Europ.png"
          />
        </div>
        <div
          class="europeDatialsDiv"
        >
          <h1
            class="euroh2eTitle"
          >
            Europe
          </h1>
          <p
            class="europePopulation"
          >
            746.4 Million
          </p>
        </div>
      </div>
      <div
        class="CountryListMainDiv"
      >
        <h2
          class="CountryListTitle"
        >
          List of Urope Countries
        </h2>
        <div
          class="CountryList"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
